### DISCUSSION TITLE
I cannot for the life of me get QT and or Cmake to work. (solved)

### DISCUSSION THREAD

**Discussion by somethingstarted** [musescore/MuseScore#16036] [Q&A]
Final Edit: it is solved. Although there is no *one* comment that solved it, it was kind of all of them that collectively helped me out of the rut. 

specs: 
-win10 pro - it's a workstation. it's plenty beefy.  specs are plenty, but if it does go the hardware route i can provide info 
-visual studio 2022,
-visual studio code version 1.74.3

posting this anyways to hopefully be of some use to somebody someday. i'm not gonna question the pros but man... why do external libraries have to be so difficult? 

i really want to contribute someday and i really did try and follow the various steps...
-here: https://musescore.org/en/handbook/developers-handbook/compilation/compile-instructions-all-platforms
-here: https://github.com/musescore/MuseScore/wiki/Compile-in-Qt-Creator
-here: https://musescore.org/en/node/323875
-here: https://musescore.org/en/node/21200
-here: https://het.as.utexas.edu/HET/Software/html/install-win.html
-here: https://youtu.be/rH2Kq2BIGVs and also
-here: https://musescore.org/en/node/317989

i've spent a week on this. i'm pretty sure at least one of these days i had version 5 installed. but who knows anymore. i've lost track of all the different things i've tried. 

![qt 04](https://user-images.githubusercontent.com/19464532/214228404-15847a57-8fbd-43ac-a311-6a9446ccd2aa.JPG)

![qt 02](https://user-images.githubusercontent.com/19464532/214227421-93ae972a-2e3f-4615-a4f8-9a04f456163d.JPG)

![qt 03](https://user-images.githubusercontent.com/19464532/214227426-36609244-275a-46e7-85a6-a19e15db8e5b.JPG)

--------------------------------------
the thing i'm going to try next is installing specifically QT v 5, which i just noticed while typing this up that the error says on cmd. so it's downloading/installing the 10gb QTv5 and i'll see what happens. 
![qt issues 01](https://user-images.githubusercontent.com/19464532/214227437-04705c94-f176-466c-a4fb-272512357082.JPG)

**Comment by cbjeukendrup** [comment]
Indeed you need Qt 5 specifically. Also, compiling in Qt Creator should be the easiest; when you try https://github.com/musescore/MuseScore/wiki/Compile-in-Qt-Creator, where does it go wrong?

**Reply by somethingstarted** [reply]
redoing the in the compiling with QT: 

first part: 
open file > find musescore source repo folder. 

click on CMakeLists.txt

choose  desktop qt 5.15.2 MSVC 2019 64bit
✓ debug
✓ release
✓ rel with db info
✓ profile
✓ min size rel

config project - so far so good. 

then...

![compile in qt - 01](https://user-images.githubusercontent.com/19464532/214243783-531a71fb-de2c-4389-8f9d-8b895a411b73.JPG)
![compile in qt - 02 no install variable](https://user-images.githubusercontent.com/19464532/214243788-29275718-c9ec-4c63-96ad-925e871c2c18.JPG)

i reconfigured the thing (below) and tried again to see if that would do anything but instal doesn't show up there. 
the first time.  - thats about as far as i can get before things start not going right 

![compile in qt - 03 nada](https://user-images.githubusercontent.com/19464532/214244028-f67a108d-1cc7-4a3f-a326-38fb8027d2e0.JPG)

-----------------------------------
edit: did some more trying and i got this error: 

Cannot run "git rev-parse --git-dir" in "C:\Users\somet\Documents\VisualStudioRepositories\MuseScoreProjects\MuseScore": 


![compile in qt - 004](https://user-images.githubusercontent.com/19464532/214281450-6e0630e7-1783-4cc7-b239-120074518bd6.JPG)
![compile in qt - 05](https://user-images.githubusercontent.com/19464532/214281453-ec1d32bd-0bbc-48eb-abf1-8be664037799.JPG)

**Reply by cbjeukendrup** [reply]
You're missing some dependencies: see https://github.com/musescore/MuseScore/wiki/Get-MuseScore%27s-source-code#adding-required-dependencies. I recommend to read through all those Wiki pages under "Compilation".

**Reply by cbjeukendrup** [reply]
Also, for that "install" variable, you're searching in the wrong spot. You should look in the CMake _options_ in the center of the screen; not in the CMake _output_ at the right.

**Reply by cbjeukendrup** [reply]
> edit: did some more trying and i got this error:
> 
> Cannot run "git rev-parse --git-dir" in "C:\Users\somet\Documents\VisualStudioRepositories\MuseScoreProjects\MuseScore":

Output in the "version control" panel is not related to building; it is related to Git operations, but you don't need to do those in Qt Creator, so don't worry about that.

**Reply by somethingstarted** [reply]
SUCCESS! 

retracing my steps: 

it had defaulted to doing some weird directory (pictured above), so i had the thought, "lemmie change that." so i clicked "browse" and it was in the {squiggly folder} which i thought was odd, so i changed it back to the MuseScore repo, and suddenly things stated loading....

![making progress - 01 - clicked browse and pointed it to my musescore fork and suddenly things started working](https://user-images.githubusercontent.com/19464532/214789373-126ef0d3-572b-42ed-8591-75bd231011f0.JPG)

when i searched for "install" in the center window you mentioned, i found **CMAKE_INSTALL_PREFIX**, Which I KNEW was a really, really good sign. so I kept on... did the steps again, fix the instaldirectory/.../.../install etc and then... 

![making progress -  02](https://user-images.githubusercontent.com/19464532/214790078-95d6bc83-aa51-481c-9eaa-657148a378f6.JPG)
![making progress - 03](https://user-images.githubusercontent.com/19464532/214790087-6e064078-2a3b-4858-b308-fb224bffa00e.JPG)
![making progress - 04 - build errors dont know if itll work now](https://user-images.githubusercontent.com/19464532/214790088-aa3f5899-1cfb-4981-b48e-01eab0206c6e.JPG)


the screenshot right above was my second to last screenshot. right after that, it said "compiled with errors, do you still want to run?" I clicked "yeah" and it did a few thingies and then MuseScore opened up. 

i can now compile musescore with my personal edits to it! - i hope to be contributing pretty soon! 

time to rock and roll i think! if not, then i'm 100x closer than i was before. 

question: if I instal in QT, i dont need to follow this compilation technique anymore, right? Would the QT comp be enough?  https://musescore.org/en/handbook/developers-handbook/compilation/compile-instructions-all-platforms

Thank you a TON! :)

**Reply by cbjeukendrup** [reply]
> question: if I instal in QT, i dont need to follow this compilation technique anymore, right? Would the QT comp be enough? https://musescore.org/en/handbook/developers-handbook/compilation/compile-instructions-all-platforms

Yes, if you can compile in Qt Creator, that is really all you need!

**Reply by cbjeukendrup** [reply]
One thing though: as the build directory you have chosen `C:/Users/somet/Documents/VisualStudioRepositories/MuseScoreMY-FORK/Musescore`. I hope that this is not the folder where you have checked out the source code... that means namely that all build files will be created in random places in the source code folder, and Git will think you want to add them to the repo (putting all these files in .gitignore is not doable). So the build directory should be different from the source directory.
